ms-ledger/main.go

```diff
@@ -1,76 +1,72 @@
-// Copyright © 2020 Intel Corporation. All rights reserved.
+// Copyright © 2022 Intel Corporation. All rights reserved.
 // SPDX-License-Identifier: BSD-3-Clause
 
 package main
 
 import (
-	"context"
-	"fmt"
+	"ms-ledger/config"
 	"ms-ledger/routes"
-	nethttp "net/http"
 	"os"
 
-	"github.com/edgexfoundry/app-functions-sdk-go/appsdk"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
 )
 
 const (
 	serviceKey = "ms-ledger"
 )
 
 func main() {
-	// Create an instance of the EdgeX SDK and initialize it.
-	edgexSdk := &appsdk.AppFunctionsSDK{ServiceKey: serviceKey}
-	if err := edgexSdk.Initialize(); err != nil {
-		edgexSdk.LoggingClient.Error(fmt.Sprintf("SDK initialization failed: %v\n", err))
-		os.Exit(-1)
+	// TODO: See https://docs.edgexfoundry.org/2.2/microservices/application/ApplicationServices/
+	//       for documentation on application services.
+	service, ok := pkg.NewAppService(serviceKey)
+	if !ok {
+		os.Exit(1)
 	}
 
-	// How to access the application's specific configuration settings.
-	appSettings := edgexSdk.ApplicationSettings()
-	if appSettings == nil {
-		edgexSdk.LoggingClient.Error("No application settings found")
-		os.Exit(-1)
+	lc := service.LoggingClient()
+
+	serviceConfig := &config.ServiceConfig{}
+	if err := service.LoadCustomConfig(serviceConfig, "AppCustom"); err != nil {
+		lc.Errorf("failed load custom configuration: %s", err.Error())
+		os.Exit(1)
+	}
+
+	if err := serviceConfig.AppCustom.Validate(); err != nil {
+		lc.Errorf("custom configuration failed validation: %s", err.Error())
+		os.Exit(1)
 	}
 
+	route := routes.NewRoute(lc, serviceConfig)
```

Comment:
Love it! Would also put the `AddRoute` calls in a `route.AddAllRoutes()` function. Will need to pass `service` to routes.NewRoute.

Reply:
Replace `serviceConfig` with the InventoryEndpoint value from App Settings.